0.8.3 breaks native builds

Steps to reproduce
1. Download the JetBrains KMP library template from https://kmp.jetbrains.com/#templateGallery
2. Modify the library's build.gradle.kts to add the uuid library
3. ./gradlew assemble

Results
```
FAILURE: Build failed with an exception.

* What went wrong:
Could not determine the dependencies of task ':library:compileKotlinLinuxX64'.
> Could not resolve all task dependencies for configuration ':library:linuxX64CompileKlibraries'.
   > Could not resolve com.benasher44:uuid:0.8.3.
     Required by:
         project :library
      > No matching variant of com.benasher44:uuid:0.8.3 was found. The consumer was configured to find a library for use during 'kotlin-api', preferably optimized for non-jvm, as well as attribute 'org.jetbrains.kotlin.platform.type' with value 'native', attribute 'org.jetbrains.kotlin.native.target' with value 'linux_x64' but:
          - Variant 'iosArm64ApiElements-published' capability com.benasher44:uuid:0.8.3 declares a library for use during 'kotlin-api', as well as attribute 'org.jetbrains.kotlin.platform.type' with value 'native':
              - Incompatible because this component declares a component, as well as attribute 'org.jetbrains.kotlin.native.target' with value 'ios_arm64' and the consumer needed a component, as well as attribute 'org.jetbrains.kotlin.native.target' with value 'linux_x64'
              - Other compatible attribute:
                  - Doesn't say anything about its target Java environment (preferred optimized for non-jvm)
          - Variant 'iosArm64MetadataElements-published' capability com.benasher44:uuid:0.8.3 declares a library, as well as attribute 'org.jetbrains.kotlin.platform.type' with value 'native':
              - Incompatible because this component declares a component for use during 'kotlin-metadata', as well as attribute 'org.jetbrains.kotlin.native.target' with value 'ios_arm64' and the consumer needed a component for use during 'kotlin-api', as well as attribute 'org.jetbrains.kotlin.native.target' with value 'linux_x64'
              - Other compatible attribute:
                  - Doesn't say anything about its target Java environment (preferred optimized for non-jvm)
          - Variant 'iosArm64SourcesElements-published' capability com.benasher44:uuid:0.8.3 declares a component, as well as attribute 'org.jetbrains.kotlin.platform.type' with value 'native':
              - Incompatible because this component declares documentation for use during 'kotlin-runtime', as well as attribute 'org.jetbrains.kotlin.native.target' with value 'ios_arm64' and the consumer needed a library for use during 'kotlin-api', as well as attribute 'org.jetbrains.kotlin.native.target' with value 'linux_x64'
              - Other compatible attribute:
                  - Doesn't say anything about its target Java environment (preferred optimized for non-jvm)
          - Variant 'iosSimulatorArm64ApiElements-published' capability com.benasher44:uuid:0.8.3 declares a library for use during 'kotlin-api', as well as attribute 'org.jetbrains.kotlin.platform.type' with value 'native':
              - Incompatible because this component declares a component, as well as attribute 'org.jetbrains.kotlin.native.target' with value 'ios_simulator_arm64' and the consumer needed a component, as well as attribute 'org.jetbrains.kotlin.native.target' with value 'linux_x64'
              - Other compatible attribute:
                  - Doesn't say anything about its target Java environment (preferred optimized for non-jvm)
          - Variant 'iosSimulatorArm64MetadataElements-published' capability com.benasher44:uuid:0.8.3 declares a library, as well as attribute 'org.jetbrains.kotlin.platform.type' with value 'native':
              - Incompatible because this component declares a component for use during 'kotlin-metadata', as well as attribute 'org.jetbrains.kotlin.native.target' with value 'ios_simulator_arm64' and the consumer needed a component for use during 'kotlin-api', as well as attribute 'org.jetbrains.kotlin.native.target' with value 'linux_x64'
              - Other compatible attribute:
                  - Doesn't say anything about its target Java environment (preferred optimized for non-jvm)
          - Variant 'iosSimulatorArm64SourcesElements-published' capability com.benasher44:uuid:0.8.3 declares a component, as well as attribute 'org.jetbrains.kotlin.platform.type' with value 'native':
              - Incompatible because this component declares documentation for use during 'kotlin-runtime', as well as attribute 'org.jetbrains.kotlin.native.target' with value 'ios_simulator_arm64' and the consumer needed a library for use during 'kotlin-api', as well as attribute 'org.jetbrains.kotlin.native.target' with value 'linux_x64'
              - Other compatible attribute:
                  - Doesn't say anything about its target Java environment (preferred optimized for non-jvm)
          - Variant 'iosX64ApiElements-published' capability com.benasher44:uuid:0.8.3 declares a library for use during 'kotlin-api', as well as attribute 'org.jetbrains.kotlin.platform.type' with value 'native':
              - Incompatible because this component declares a component, as well as attribute 'org.jetbrains.kotlin.native.target' with value 'ios_x64' and the consumer needed a component, as well as attribute 'org.jetbrains.kotlin.native.target' with value 'linux_x64'
              - Other compatible attribute:
                  - Doesn't say anything about its target Java environment (preferred optimized for non-jvm)
          - Variant 'iosX64MetadataElements-published' capability com.benasher44:uuid:0.8.3 declares a library, as well as attribute 'org.jetbrains.kotlin.platform.type' with value 'native':
              - Incompatible because this component declares a component for use during 'kotlin-metadata', as well as attribute 'org.jetbrains.kotlin.native.target' with value 'ios_x64' and the consumer needed a component for use during 'kotlin-api', as well as attribute 'org.jetbrains.kotlin.native.target' with value 'linux_x64'
              - Other compatible attribute:
                  - Doesn't say anything about its target Java environment (preferred optimized for non-jvm)
          - Variant 'iosX64SourcesElements-published' capability com.benasher44:uuid:0.8.3 declares a component, as well as attribute 'org.jetbrains.kotlin.platform.type' with value 'native':
              - Incompatible because this component declares documentation for use during 'kotlin-runtime', as well as attribute 'org.jetbrains.kotlin.native.target' with value 'ios_x64' and the consumer needed a library for use during 'kotlin-api', as well as attribute 'org.jetbrains.kotlin.native.target' with value 'linux_x64'
              - Other compatible attribute:
                  - Doesn't say anything about its target Java environment (preferred optimized for non-jvm)
          - Variant 'jsApiElements-published' capability com.benasher44:uuid:0.8.3 declares a library for use during 'kotlin-api':
              - Incompatible because this component declares a component, as well as attribute 'org.jetbrains.kotlin.platform.type' with value 'js' and the consumer needed a component, as well as attribute 'org.jetbrains.kotlin.platform.type' with value 'native'
              - Other compatible attributes:
                  - Doesn't say anything about its target Java environment (preferred optimized for non-jvm)
                  - Doesn't say anything about org.jetbrains.kotlin.native.target (required 'linux_x64')
          - Variant 'jsRuntimeElements-published' capability com.benasher44:uuid:0.8.3 declares a library:
              - Incompatible because this component declares a component for use during 'kotlin-runtime', as well as attribute 'org.jetbrains.kotlin.platform.type' with value 'js' and the consumer needed a component for use during 'kotlin-api', as well as attribute 'org.jetbrains.kotlin.platform.type' with value 'native'
              - Other compatible attributes:
                  - Doesn't say anything about its target Java environment (preferred optimized for non-jvm)
                  - Doesn't say anything about org.jetbrains.kotlin.native.target (required 'linux_x64')
          - Variant 'jsSourcesElements-published' capability com.benasher44:uuid:0.8.3:
              - Incompatible because this component declares documentation for use during 'kotlin-runtime', as well as attribute 'org.jetbrains.kotlin.platform.type' with value 'js' and the consumer needed a library for use during 'kotlin-api', as well as attribute 'org.jetbrains.kotlin.platform.type' with value 'native'
              - Other compatible attributes:
                  - Doesn't say anything about its target Java environment (preferred optimized for non-jvm)
                  - Doesn't say anything about org.jetbrains.kotlin.native.target (required 'linux_x64')
          - Variant 'jvmApiElements-published' capability com.benasher44:uuid:0.8.3 declares a library for use during compile-time:
              - Incompatible because this component declares a component, as well as attribute 'org.jetbrains.kotlin.platform.type' with value 'jvm' and the consumer needed a component, as well as attribute 'org.jetbrains.kotlin.platform.type' with value 'native'
              - Other compatible attributes:
                  - Doesn't say anything about its target Java environment (preferred optimized for non-jvm)
                  - Doesn't say anything about org.jetbrains.kotlin.native.target (required 'linux_x64')
          - Variant 'jvmRuntimeElements-published' capability com.benasher44:uuid:0.8.3 declares a library for use during runtime:
              - Incompatible because this component declares a component, as well as attribute 'org.jetbrains.kotlin.platform.type' with value 'jvm' and the consumer needed a component, as well as attribute 'org.jetbrains.kotlin.platform.type' with value 'native'
              - Other compatible attributes:
                  - Doesn't say anything about its target Java environment (preferred optimized for non-jvm)
                  - Doesn't say anything about org.jetbrains.kotlin.native.target (required 'linux_x64')
          - Variant 'jvmSourcesElements-published' capability com.benasher44:uuid:0.8.3 declares a component for use during runtime:
              - Incompatible because this component declares documentation, as well as attribute 'org.jetbrains.kotlin.platform.type' with value 'jvm' and the consumer needed a library, as well as attribute 'org.jetbrains.kotlin.platform.type' with value 'native'
              - Other compatible attributes:
                  - Doesn't say anything about its target Java environment (preferred optimized for non-jvm)
                  - Doesn't say anything about org.jetbrains.kotlin.native.target (required 'linux_x64')
          - Variant 'macosArm64ApiElements-published' capability com.benasher44:uuid:0.8.3 declares a library for use during 'kotlin-api', as well as attribute 'org.jetbrains.kotlin.platform.type' with value 'native':
              - Incompatible because this component declares a component, as well as attribute 'org.jetbrains.kotlin.native.target' with value 'macos_arm64' and the consumer needed a component, as well as attribute 'org.jetbrains.kotlin.native.target' with value 'linux_x64'
              - Other compatible attribute:
                  - Doesn't say anything about its target Java environment (preferred optimized for non-jvm)
          - Variant 'macosArm64MetadataElements-published' capability com.benasher44:uuid:0.8.3 declares a library, as well as attribute 'org.jetbrains.kotlin.platform.type' with value 'native':
              - Incompatible because this component declares a component for use during 'kotlin-metadata', as well as attribute 'org.jetbrains.kotlin.native.target' with value 'macos_arm64' and the consumer needed a component for use during 'kotlin-api', as well as attribute 'org.jetbrains.kotlin.native.target' with value 'linux_x64'
              - Other compatible attribute:
                  - Doesn't say anything about its target Java environment (preferred optimized for non-jvm)
          - Variant 'macosArm64SourcesElements-published' capability com.benasher44:uuid:0.8.3 declares a component, as well as attribute 'org.jetbrains.kotlin.platform.type' with value 'native':
              - Incompatible because this component declares documentation for use during 'kotlin-runtime', as well as attribute 'org.jetbrains.kotlin.native.target' with value 'macos_arm64' and the consumer needed a library for use during 'kotlin-api', as well as attribute 'org.jetbrains.kotlin.native.target' with value 'linux_x64'
              - Other compatible attribute:
                  - Doesn't say anything about its target Java environment (preferred optimized for non-jvm)
          - Variant 'macosX64ApiElements-published' capability com.benasher44:uuid:0.8.3 declares a library for use during 'kotlin-api', as well as attribute 'org.jetbrains.kotlin.platform.type' with value 'native':
              - Incompatible because this component declares a component, as well as attribute 'org.jetbrains.kotlin.native.target' with value 'macos_x64' and the consumer needed a component, as well as attribute 'org.jetbrains.kotlin.native.target' with value 'linux_x64'
              - Other compatible attribute:
                  - Doesn't say anything about its target Java environment (preferred optimized for non-jvm)
          - Variant 'macosX64MetadataElements-published' capability com.benasher44:uuid:0.8.3 declares a library, as well as attribute 'org.jetbrains.kotlin.platform.type' with value 'native':
              - Incompatible because this component declares a component for use during 'kotlin-metadata', as well as attribute 'org.jetbrains.kotlin.native.target' with value 'macos_x64' and the consumer needed a component for use during 'kotlin-api', as well as attribute 'org.jetbrains.kotlin.native.target' with value 'linux_x64'
              - Other compatible attribute:
                  - Doesn't say anything about its target Java environment (preferred optimized for non-jvm)
          - Variant 'macosX64SourcesElements-published' capability com.benasher44:uuid:0.8.3 declares a component, as well as attribute 'org.jetbrains.kotlin.platform.type' with value 'native':
              - Incompatible because this component declares documentation for use during 'kotlin-runtime', as well as attribute 'org.jetbrains.kotlin.native.target' with value 'macos_x64' and the consumer needed a library for use during 'kotlin-api', as well as attribute 'org.jetbrains.kotlin.native.target' with value 'linux_x64'
              - Other compatible attribute:
                  - Doesn't say anything about its target Java environment (preferred optimized for non-jvm)
          - Variant 'metadataApiElements' capability com.benasher44:uuid:0.8.3 declares a library:
              - Incompatible because this component declares a component for use during 'kotlin-metadata', as well as attribute 'org.jetbrains.kotlin.platform.type' with value 'common' and the consumer needed a component for use during 'kotlin-api', as well as attribute 'org.jetbrains.kotlin.platform.type' with value 'native'
              - Other compatible attributes:
                  - Doesn't say anything about its target Java environment (preferred optimized for non-jvm)
                  - Doesn't say anything about org.jetbrains.kotlin.native.target (required 'linux_x64')
          - Variant 'metadataSourcesElements' capability com.benasher44:uuid:0.8.3:
              - Incompatible because this component declares documentation for use during 'kotlin-runtime', as well as attribute 'org.jetbrains.kotlin.platform.type' with value 'common' and the consumer needed a library for use during 'kotlin-api', as well as attribute 'org.jetbrains.kotlin.platform.type' with value 'native'
              - Other compatible attributes:
                  - Doesn't say anything about its target Java environment (preferred optimized for non-jvm)
                  - Doesn't say anything about org.jetbrains.kotlin.native.target (required 'linux_x64')
          - Variant 'mingwX64ApiElements-published' capability com.benasher44:uuid:0.8.3 declares a library for use during 'kotlin-api', as well as attribute 'org.jetbrains.kotlin.platform.type' with value 'native':
              - Incompatible because this component declares a component, as well as attribute 'org.jetbrains.kotlin.native.target' with value 'mingw_x64' and the consumer needed a component, as well as attribute 'org.jetbrains.kotlin.native.target' with value 'linux_x64'
              - Other compatible attribute:
                  - Doesn't say anything about its target Java environment (preferred optimized for non-jvm)
          - Variant 'mingwX64SourcesElements-published' capability com.benasher44:uuid:0.8.3 declares a component, as well as attribute 'org.jetbrains.kotlin.platform.type' with value 'native':
              - Incompatible because this component declares documentation for use during 'kotlin-runtime', as well as attribute 'org.jetbrains.kotlin.native.target' with value 'mingw_x64' and the consumer needed a library for use during 'kotlin-api', as well as attribute 'org.jetbrains.kotlin.native.target' with value 'linux_x64'
              - Other compatible attribute:
                  - Doesn't say anything about its target Java environment (preferred optimized for non-jvm)
          - Variant 'tvosArm64ApiElements-published' capability com.benasher44:uuid:0.8.3 declares a library for use during 'kotlin-api', as well as attribute 'org.jetbrains.kotlin.platform.type' with value 'native':
              - Incompatible because this component declares a component, as well as attribute 'org.jetbrains.kotlin.native.target' with value 'tvos_arm64' and the consumer needed a component, as well as attribute 'org.jetbrains.kotlin.native.target' with value 'linux_x64'
              - Other compatible attribute:
                  - Doesn't say anything about its target Java environment (preferred optimized for non-jvm)
          - Variant 'tvosArm64MetadataElements-published' capability com.benasher44:uuid:0.8.3 declares a library, as well as attribute 'org.jetbrains.kotlin.platform.type' with value 'native':
              - Incompatible because this component declares a component for use during 'kotlin-metadata', as well as attribute 'org.jetbrains.kotlin.native.target' with value 'tvos_arm64' and the consumer needed a component for use during 'kotlin-api', as well as attribute 'org.jetbrains.kotlin.native.target' with value 'linux_x64'
              - Other compatible attribute:
                  - Doesn't say anything about its target Java environment (preferred optimized for non-jvm)
          - Variant 'tvosArm64SourcesElements-published' capability com.benasher44:uuid:0.8.3 declares a component, as well as attribute 'org.jetbrains.kotlin.platform.type' with value 'native':
              - Incompatible because this component declares documentation for use during 'kotlin-runtime', as well as attribute 'org.jetbrains.kotlin.native.target' with value 'tvos_arm64' and the consumer needed a library for use during 'kotlin-api', as well as attribute 'org.jetbrains.kotlin.native.target' with value 'linux_x64'
              - Other compatible attribute:
                  - Doesn't say anything about its target Java environment (preferred optimized for non-jvm)
          - Variant 'tvosSimulatorArm64ApiElements-published' capability com.benasher44:uuid:0.8.3 declares a library for use during 'kotlin-api', as well as attribute 'org.jetbrains.kotlin.platform.type' with value 'native':
              - Incompatible because this component declares a component, as well as attribute 'org.jetbrains.kotlin.native.target' with value 'tvos_simulator_arm64' and the consumer needed a component, as well as attribute 'org.jetbrains.kotlin.native.target' with value 'linux_x64'
              - Other compatible attribute:
                  - Doesn't say anything about its target Java environment (preferred optimized for non-jvm)
          - Variant 'tvosSimulatorArm64MetadataElements-published' capability com.benasher44:uuid:0.8.3 declares a library, as well as attribute 'org.jetbrains.kotlin.platform.type' with value 'native':
              - Incompatible because this component declares a component for use during 'kotlin-metadata', as well as attribute 'org.jetbrains.kotlin.native.target' with value 'tvos_simulator_arm64' and the consumer needed a component for use during 'kotlin-api', as well as attribute 'org.jetbrains.kotlin.native.target' with value 'linux_x64'
              - Other compatible attribute:
                  - Doesn't say anything about its target Java environment (preferred optimized for non-jvm)
          - Variant 'tvosSimulatorArm64SourcesElements-published' capability com.benasher44:uuid:0.8.3 declares a component, as well as attribute 'org.jetbrains.kotlin.platform.type' with value 'native':
              - Incompatible because this component declares documentation for use during 'kotlin-runtime', as well as attribute 'org.jetbrains.kotlin.native.target' with value 'tvos_simulator_arm64' and the consumer needed a library for use during 'kotlin-api', as well as attribute 'org.jetbrains.kotlin.native.target' with value 'linux_x64'
              - Other compatible attribute:
                  - Doesn't say anything about its target Java environment (preferred optimized for non-jvm)
          - Variant 'tvosX64ApiElements-published' capability com.benasher44:uuid:0.8.3 declares a library for use during 'kotlin-api', as well as attribute 'org.jetbrains.kotlin.platform.type' with value 'native':
              - Incompatible because this component declares a component, as well as attribute 'org.jetbrains.kotlin.native.target' with value 'tvos_x64' and the consumer needed a component, as well as attribute 'org.jetbrains.kotlin.native.target' with value 'linux_x64'
              - Other compatible attribute:
                  - Doesn't say anything about its target Java environment (preferred optimized for non-jvm)
          - Variant 'tvosX64MetadataElements-published' capability com.benasher44:uuid:0.8.3 declares a library, as well as attribute 'org.jetbrains.kotlin.platform.type' with value 'native':
              - Incompatible because this component declares a component for use during 'kotlin-metadata', as well as attribute 'org.jetbrains.kotlin.native.target' with value 'tvos_x64' and the consumer needed a component for use during 'kotlin-api', as well as attribute 'org.jetbrains.kotlin.native.target' with value 'linux_x64'
              - Other compatible attribute:
                  - Doesn't say anything about its target Java environment (preferred optimized for non-jvm)
          - Variant 'tvosX64SourcesElements-published' capability com.benasher44:uuid:0.8.3 declares a component, as well as attribute 'org.jetbrains.kotlin.platform.type' with value 'native':
              - Incompatible because this component declares documentation for use during 'kotlin-runtime', as well as attribute 'org.jetbrains.kotlin.native.target' with value 'tvos_x64' and the consumer needed a library for use during 'kotlin-api', as well as attribute 'org.jetbrains.kotlin.native.target' with value 'linux_x64'
              - Other compatible attribute:
                  - Doesn't say anything about its target Java environment (preferred optimized for non-jvm)
          - Variant 'wasmJsApiElements-published' capability com.benasher44:uuid:0.8.3 declares a library for use during 'kotlin-api':
              - Incompatible because this component declares a component, as well as attribute 'org.jetbrains.kotlin.platform.type' with value 'wasm' and the consumer needed a component, as well as attribute 'org.jetbrains.kotlin.platform.type' with value 'native'
              - Other compatible attributes:
                  - Doesn't say anything about its target Java environment (preferred optimized for non-jvm)
                  - Doesn't say anything about org.jetbrains.kotlin.native.target (required 'linux_x64')
          - Variant 'wasmJsRuntimeElements-published' capability com.benasher44:uuid:0.8.3 declares a library:
              - Incompatible because this component declares a component for use during 'kotlin-runtime', as well as attribute 'org.jetbrains.kotlin.platform.type' with value 'wasm' and the consumer needed a component for use during 'kotlin-api', as well as attribute 'org.jetbrains.kotlin.platform.type' with value 'native'
              - Other compatible attributes:
                  - Doesn't say anything about its target Java environment (preferred optimized for non-jvm)
                  - Doesn't say anything about org.jetbrains.kotlin.native.target (required 'linux_x64')
          - Variant 'wasmJsSourcesElements-published' capability com.benasher44:uuid:0.8.3:
              - Incompatible because this component declares documentation for use during 'kotlin-runtime', as well as attribute 'org.jetbrains.kotlin.platform.type' with value 'wasm' and the consumer needed a library for use during 'kotlin-api', as well as attribute 'org.jetbrains.kotlin.platform.type' with value 'native'
              - Other compatible attributes:
                  - Doesn't say anything about its target Java environment (preferred optimized for non-jvm)
                  - Doesn't say anything about org.jetbrains.kotlin.native.target (required 'linux_x64')
          - Variant 'watchosArm32ApiElements-published' capability com.benasher44:uuid:0.8.3 declares a library for use during 'kotlin-api', as well as attribute 'org.jetbrains.kotlin.platform.type' with value 'native':
              - Incompatible because this component declares a component, as well as attribute 'org.jetbrains.kotlin.native.target' with value 'watchos_arm32' and the consumer needed a component, as well as attribute 'org.jetbrains.kotlin.native.target' with value 'linux_x64'
              - Other compatible attribute:
                  - Doesn't say anything about its target Java environment (preferred optimized for non-jvm)
          - Variant 'watchosArm32MetadataElements-published' capability com.benasher44:uuid:0.8.3 declares a library, as well as attribute 'org.jetbrains.kotlin.platform.type' with value 'native':
              - Incompatible because this component declares a component for use during 'kotlin-metadata', as well as attribute 'org.jetbrains.kotlin.native.target' with value 'watchos_arm32' and the consumer needed a component for use during 'kotlin-api', as well as attribute 'org.jetbrains.kotlin.native.target' with value 'linux_x64'
              - Other compatible attribute:
                  - Doesn't say anything about its target Java environment (preferred optimized for non-jvm)
          - Variant 'watchosArm32SourcesElements-published' capability com.benasher44:uuid:0.8.3 declares a component, as well as attribute 'org.jetbrains.kotlin.platform.type' with value 'native':
              - Incompatible because this component declares documentation for use during 'kotlin-runtime', as well as attribute 'org.jetbrains.kotlin.native.target' with value 'watchos_arm32' and the consumer needed a library for use during 'kotlin-api', as well as attribute 'org.jetbrains.kotlin.native.target' with value 'linux_x64'
              - Other compatible attribute:
                  - Doesn't say anything about its target Java environment (preferred optimized for non-jvm)
          - Variant 'watchosArm64ApiElements-published' capability com.benasher44:uuid:0.8.3 declares a library for use during 'kotlin-api', as well as attribute 'org.jetbrains.kotlin.platform.type' with value 'native':
              - Incompatible because this component declares a component, as well as attribute 'org.jetbrains.kotlin.native.target' with value 'watchos_arm64' and the consumer needed a component, as well as attribute 'org.jetbrains.kotlin.native.target' with value 'linux_x64'
              - Other compatible attribute:
                  - Doesn't say anything about its target Java environment (preferred optimized for non-jvm)
          - Variant 'watchosArm64MetadataElements-published' capability com.benasher44:uuid:0.8.3 declares a library, as well as attribute 'org.jetbrains.kotlin.platform.type' with value 'native':
              - Incompatible because this component declares a component for use during 'kotlin-metadata', as well as attribute 'org.jetbrains.kotlin.native.target' with value 'watchos_arm64' and the consumer needed a component for use during 'kotlin-api', as well as attribute 'org.jetbrains.kotlin.native.target' with value 'linux_x64'
              - Other compatible attribute:
                  - Doesn't say anything about its target Java environment (preferred optimized for non-jvm)
          - Variant 'watchosArm64SourcesElements-published' capability com.benasher44:uuid:0.8.3 declares a component, as well as attribute 'org.jetbrains.kotlin.platform.type' with value 'native':
              - Incompatible because this component declares documentation for use during 'kotlin-runtime', as well as attribute 'org.jetbrains.kotlin.native.target' with value 'watchos_arm64' and the consumer needed a library for use during 'kotlin-api', as well as attribute 'org.jetbrains.kotlin.native.target' with value 'linux_x64'
              - Other compatible attribute:
                  - Doesn't say anything about its target Java environment (preferred optimized for non-jvm)
          - Variant 'watchosDeviceArm64ApiElements-published' capability com.benasher44:uuid:0.8.3 declares a library for use during 'kotlin-api', as well as attribute 'org.jetbrains.kotlin.platform.type' with value 'native':
              - Incompatible because this component declares a component, as well as attribute 'org.jetbrains.kotlin.native.target' with value 'watchos_device_arm64' and the consumer needed a component, as well as attribute 'org.jetbrains.kotlin.native.target' with value 'linux_x64'
              - Other compatible attribute:
                  - Doesn't say anything about its target Java environment (preferred optimized for non-jvm)
          - Variant 'watchosDeviceArm64MetadataElements-published' capability com.benasher44:uuid:0.8.3 declares a library, as well as attribute 'org.jetbrains.kotlin.platform.type' with value 'native':
              - Incompatible because this component declares a component for use during 'kotlin-metadata', as well as attribute 'org.jetbrains.kotlin.native.target' with value 'watchos_device_arm64' and the consumer needed a component for use during 'kotlin-api', as well as attribute 'org.jetbrains.kotlin.native.target' with value 'linux_x64'
              - Other compatible attribute:
                  - Doesn't say anything about its target Java environment (preferred optimized for non-jvm)
          - Variant 'watchosDeviceArm64SourcesElements-published' capability com.benasher44:uuid:0.8.3 declares a component, as well as attribute 'org.jetbrains.kotlin.platform.type' with value 'native':
              - Incompatible because this component declares documentation for use during 'kotlin-runtime', as well as attribute 'org.jetbrains.kotlin.native.target' with value 'watchos_device_arm64' and the consumer needed a library for use during 'kotlin-api', as well as attribute 'org.jetbrains.kotlin.native.target' with value 'linux_x64'
              - Other compatible attribute:
                  - Doesn't say anything about its target Java environment (preferred optimized for non-jvm)
          - Variant 'watchosSimulatorArm64ApiElements-published' capability com.benasher44:uuid:0.8.3 declares a library for use during 'kotlin-api', as well as attribute 'org.jetbrains.kotlin.platform.type' with value 'native':
              - Incompatible because this component declares a component, as well as attribute 'org.jetbrains.kotlin.native.target' with value 'watchos_simulator_arm64' and the consumer needed a component, as well as attribute 'org.jetbrains.kotlin.native.target' with value 'linux_x64'
              - Other compatible attribute:
                  - Doesn't say anything about its target Java environment (preferred optimized for non-jvm)
          - Variant 'watchosSimulatorArm64MetadataElements-published' capability com.benasher44:uuid:0.8.3 declares a library, as well as attribute 'org.jetbrains.kotlin.platform.type' with value 'native':
              - Incompatible because this component declares a component for use during 'kotlin-metadata', as well as attribute 'org.jetbrains.kotlin.native.target' with value 'watchos_simulator_arm64' and the consumer needed a component for use during 'kotlin-api', as well as attribute 'org.jetbrains.kotlin.native.target' with value 'linux_x64'
              - Other compatible attribute:
                  - Doesn't say anything about its target Java environment (preferred optimized for non-jvm)
          - Variant 'watchosSimulatorArm64SourcesElements-published' capability com.benasher44:uuid:0.8.3 declares a component, as well as attribute 'org.jetbrains.kotlin.platform.type' with value 'native':
              - Incompatible because this component declares documentation for use during 'kotlin-runtime', as well as attribute 'org.jetbrains.kotlin.native.target' with value 'watchos_simulator_arm64' and the consumer needed a library for use during 'kotlin-api', as well as attribute 'org.jetbrains.kotlin.native.target' with value 'linux_x64'
              - Other compatible attribute:
                  - Doesn't say anything about its target Java environment (preferred optimized for non-jvm)
          - Variant 'watchosX64ApiElements-published' capability com.benasher44:uuid:0.8.3 declares a library for use during 'kotlin-api', as well as attribute 'org.jetbrains.kotlin.platform.type' with value 'native':
              - Incompatible because this component declares a component, as well as attribute 'org.jetbrains.kotlin.native.target' with value 'watchos_x64' and the consumer needed a component, as well as attribute 'org.jetbrains.kotlin.native.target' with value 'linux_x64'
              - Other compatible attribute:
                  - Doesn't say anything about its target Java environment (preferred optimized for non-jvm)
          - Variant 'watchosX64MetadataElements-published' capability com.benasher44:uuid:0.8.3 declares a library, as well as attribute 'org.jetbrains.kotlin.platform.type' with value 'native':
              - Incompatible because this component declares a component for use during 'kotlin-metadata', as well as attribute 'org.jetbrains.kotlin.native.target' with value 'watchos_x64' and the consumer needed a component for use during 'kotlin-api', as well as attribute 'org.jetbrains.kotlin.native.target' with value 'linux_x64'
              - Other compatible attribute:
                  - Doesn't say anything about its target Java environment (preferred optimized for non-jvm)
          - Variant 'watchosX64SourcesElements-published' capability com.benasher44:uuid:0.8.3 declares a component, as well as attribute 'org.jetbrains.kotlin.platform.type' with value 'native':
              - Incompatible because this component declares documentation for use during 'kotlin-runtime', as well as attribute 'org.jetbrains.kotlin.native.target' with value 'watchos_x64' and the consumer needed a library for use during 'kotlin-api', as well as attribute 'org.jetbrains.kotlin.native.target' with value 'linux_x64'
              - Other compatible attribute:
                  - Doesn't say anything about its target Java environment (preferred optimized for non-jvm)

* Try:
> Review the variant matching algorithm at https://docs.gradle.org/8.6/userguide/variant_attributes.html#sec:abm_algorithm.
> No matching variant errors are explained in more detail at https://docs.gradle.org/8.6/userguide/variant_model.html#sub:variant-no-match.
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 9.0.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/8.6/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -5,10 +5,14 @@ All notable changes to this project will be documented in this file.
 The format is based on [Keep a Changelog](https://keepachangelog.com/), and this
 project adheres to [Semantic Versioning](https://semver.org/).
 
+## [0.8.4] - 2024-03-19
+### Changed
+- Fix Kotlin Metadata (#157)
+
 ## [0.8.3] - 2024-03-16
 ### Changed
-- Bump kotlin to 1.9.23
-- Add wasmWasi support
+- Bump kotlin to 1.9.23 (#154)
+- Add wasmWasi support (#154)
 
 ## [0.8.2] - 2023-11-10
 ### Changed

--- a/build.gradle.kts
+++ b/build.gradle.kts
@@ -46,31 +46,30 @@ kotlin {
         wasmJs {
             d8()
         }
-        if (HostManager.hostIsMac) {
-            macosX64()
-            macosArm64()
-            iosX64()
-            iosArm64()
-            iosSimulatorArm64()
-            watchosArm32()
-            watchosArm64()
-            watchosX64()
-            watchosSimulatorArm64()
-            watchosDeviceArm64()
-            tvosArm64()
-            tvosX64()
-            tvosSimulatorArm64()
+        macosX64()
+        macosArm64()
+        iosX64()
+        iosArm64()
+        iosSimulatorArm64()
+        watchosArm32()
+        watchosArm64()
+        watchosX64()
+        watchosSimulatorArm64()
+        watchosDeviceArm64()
+        tvosArm64()
+        tvosX64()
+        tvosSimulatorArm64()
+        mingwX64 {
+            binaries.findTest(DEBUG)!!.linkerOpts = mutableListOf("-Wl,--subsystem,windows")
         }
-        if (HostManager.hostIsMingw || HostManager.hostIsMac) {
-            mingwX64 {
-                binaries.findTest(DEBUG)!!.linkerOpts = mutableListOf("-Wl,--subsystem,windows")
-            }
-        }
-        if (HostManager.hostIsLinux) {
-            linuxX64()
-            linuxArm64()
-            wasmWasi {
-                nodejs()
+        linuxX64()
+        linuxArm64()
+        wasmWasi {
+            nodejs {
+                testTask {
+                    // skip on windows, since there doesn't seem to be a canary node version for use on windows
+                    this.enabled = !HostManager.hostIsMingw
+                }
             }
         }
     }
@@ -98,68 +97,59 @@ kotlin {
         val nix64Test by creating { dependsOn(nativeTest) }
         val nix32Main by creating { dependsOn(nativeMain) }
         val nix32Test by creating { dependsOn(nativeTest) }
-
-        if (HostManager.hostIsMac) {
-            val appleMain by creating { dependsOn(nativeMain) }
-            val appleTest by creating { dependsOn(nativeTest) }
-            val apple64Main by creating {
-                dependsOn(appleMain)
-                dependsOn(nix64Main)
-            }
-            val apple64Test by creating {
-                dependsOn(appleTest)
-                dependsOn(nix64Test)
-            }
-            val apple32Main by creating {
-                dependsOn(appleMain)
-                dependsOn(nix32Main)
-            }
-            val apple32Test by creating {
-                dependsOn(appleTest)
-                dependsOn(nix32Test)
-            }
-            val iosX64Main by getting { dependsOn(apple64Main) }
-            val iosX64Test by getting { dependsOn(apple64Test) }
-            val iosArm64Main by getting { dependsOn(apple64Main) }
-            val iosArm64Test by getting { dependsOn(apple64Test) }
-            val macosX64Main by getting { dependsOn(apple64Main) }
-            val macosX64Test by getting { dependsOn(apple64Test) }
-            val macosArm64Main by getting { dependsOn(apple64Main) }
-            val macosArm64Test by getting { dependsOn(apple64Test) }
-            val iosSimulatorArm64Main by getting { dependsOn(apple64Main) }
-            val iosSimulatorArm64Test by getting { dependsOn(apple64Test) }
-            val watchosArm32Main by getting { dependsOn(apple32Main) }
-            val watchosArm32Test by getting { dependsOn(apple32Test) }
-            val watchosArm64Main by getting { dependsOn(apple64Main) }
-            val watchosArm64Test by getting { dependsOn(apple64Test) }
-            val watchosX64Main by getting { dependsOn(apple64Main) }
-            val watchosX64Test by getting { dependsOn(apple64Test) }
-            val watchosSimulatorArm64Main by getting { dependsOn(apple64Main) }
-            val watchosSimulatorArm64Test by getting { dependsOn(apple64Test) }
-            val watchosDeviceArm64Main by getting { dependsOn(apple64Main) }
-            val watchosDeviceArm64Test by getting { dependsOn(apple64Test) }
-            val tvosArm64Main by getting { dependsOn(apple64Main) }
-            val tvosArm64Test by getting { dependsOn(apple64Test) }
-            val tvosX64Main by getting { dependsOn(apple64Main) }
-            val tvosX64Test by getting { dependsOn(apple64Test) }
-            val tvosSimulatorArm64Main by getting { dependsOn(apple64Main) }
-            val tvosSimulatorArm64Test by getting { dependsOn(apple64Test) }
+        val appleMain by creating { dependsOn(nativeMain) }
+        val appleTest by creating { dependsOn(nativeTest) }
+        val apple64Main by creating {
+            dependsOn(appleMain)
+            dependsOn(nix64Main)
         }
-
-        if (HostManager.hostIsMingw || HostManager.hostIsMac) {
-            val mingwMain by creating { dependsOn(nativeMain) }
-            val mingwTest by creating { dependsOn(nativeTest) }
-            val mingwX64Main by getting { dependsOn(mingwMain) }
-            val mingwX64Test by getting { dependsOn(mingwTest) }
+        val apple64Test by creating {
+            dependsOn(appleTest)
+            dependsOn(nix64Test)
         }
-
-        if (HostManager.hostIsLinux) {
-            val linuxX64Main by getting { dependsOn(nix64Main) }
-            val linuxX64Test by getting { dependsOn(nix64Test) }
-            val linuxArm64Main by getting { dependsOn(nix64Main) }
-            val linuxArm64Test by getting { dependsOn(nix64Test) }
-            val wasmWasiMain by getting { dependsOn(nonJvmMain) }
+        val apple32Main by creating {
+            dependsOn(appleMain)
+            dependsOn(nix32Main)
         }
+        val apple32Test by creating {
+            dependsOn(appleTest)
+            dependsOn(nix32Test)
+        }
+        val iosX64Main by getting { dependsOn(apple64Main) }
+        val iosX64Test by getting { dependsOn(apple64Test) }
+        val iosArm64Main by getting { dependsOn(apple64Main) }
+        val iosArm64Test by getting { dependsOn(apple64Test) }
+        val macosX64Main by getting { dependsOn(apple64Main) }
+        val macosX64Test by getting { dependsOn(apple64Test) }
+        val macosArm64Main by getting { dependsOn(apple64Main) }
+        val macosArm64Test by getting { dependsOn(apple64Test) }
+        val iosSimulatorArm64Main by getting { dependsOn(apple64Main) }
+        val iosSimulatorArm64Test by getting { dependsOn(apple64Test) }
+        val watchosArm32Main by getting { dependsOn(apple32Main) }
+        val watchosArm32Test by getting { dependsOn(apple32Test) }
+        val watchosArm64Main by getting { dependsOn(apple64Main) }
+        val watchosArm64Test by getting { dependsOn(apple64Test) }
+        val watchosX64Main by getting { dependsOn(apple64Main) }
+        val watchosX64Test by getting { dependsOn(apple64Test) }
+        val watchosSimulatorArm64Main by getting { dependsOn(apple64Main) }
+        val watchosSimulatorArm64Test by getting { dependsOn(apple64Test) }
+        val watchosDeviceArm64Main by getting { dependsOn(apple64Main) }
+        val watchosDeviceArm64Test by getting { dependsOn(apple64Test) }
+        val tvosArm64Main by getting { dependsOn(apple64Main) }
+        val tvosArm64Test by getting { dependsOn(apple64Test) }
+        val tvosX64Main by getting { dependsOn(apple64Main) }
+        val tvosX64Test by getting { dependsOn(apple64Test) }
+        val tvosSimulatorArm64Main by getting { dependsOn(apple64Main) }
+        val tvosSimulatorArm64Test by getting { dependsOn(apple64Test) }
+        val mingwMain by creating { dependsOn(nativeMain) }
+        val mingwTest by creating { dependsOn(nativeTest) }
+        val mingwX64Main by getting { dependsOn(mingwMain) }
+        val mingwX64Test by getting { dependsOn(mingwTest) }
+        val linuxX64Main by getting { dependsOn(nix64Main) }
+        val linuxX64Test by getting { dependsOn(nix64Test) }
+        val linuxArm64Main by getting { dependsOn(nix64Main) }
+        val linuxArm64Test by getting { dependsOn(nix64Test) }
+        val wasmWasiMain by getting { dependsOn(nonJvmMain) }
     }
 }
 
@@ -171,7 +161,8 @@ tasks.withType<KotlinNativeCompile>().configureEach {
     compilerOptions.freeCompilerArgs.add("-opt-in=kotlinx.cinterop.ExperimentalForeignApi")
 }
 
-if (HostManager.hostIsLinux) {
+if (!HostManager.hostIsMingw) {
+    // doesn't seem to be a canary version for use on windows
     plugins.withType<org.jetbrains.kotlin.gradle.targets.js.nodejs.NodeJsRootPlugin> {
         the<org.jetbrains.kotlin.gradle.targets.js.nodejs.NodeJsRootExtension>().download = true
         the<org.jetbrains.kotlin.gradle.targets.js.nodejs.NodeJsRootExtension>().nodeVersion =
@@ -179,10 +170,10 @@ if (HostManager.hostIsLinux) {
         the<org.jetbrains.kotlin.gradle.targets.js.nodejs.NodeJsRootExtension>().nodeDownloadBaseUrl =
             "https://nodejs.org/download/v8-canary"
     }
+}
 
-    tasks.withType<org.jetbrains.kotlin.gradle.targets.js.npm.tasks.KotlinNpmInstallTask>().configureEach {
-        args.add("--ignore-engines")
-    }
+tasks.withType<org.jetbrains.kotlin.gradle.targets.js.npm.tasks.KotlinNpmInstallTask>().configureEach {
+    args.add("--ignore-engines")
 }
 
 val ktlintConfig by configurations.creating

--- a/gradle.properties
+++ b/gradle.properties
@@ -5,7 +5,7 @@ kotlin.code.style=official
 kotlin.incremental=true
 
 GROUP=com.benasher44
-VERSION=0.8.3
+VERSION=0.8.4
 
 POM_URL=https://github.com/benasher44/uuid/
 POM_SCM_URL=https://github.com/benasher44/uuid/

